### DISCUSSION TITLE
How to use Sender Id or Caller Id instead of phone number?

### DISCUSSION THREAD

**Discussion by lujamaharjan** [python-smpplib/python-smpplib#166] [Q&A] [answered]
Hi, @eigenein @podshumok  @baverman 
How can we use sender id to recipients of SMS instead of our phone number (source_address)?
When we receive transaction alerts from banks and other institutions we get BANK_ALERT like text instead
of their mobile number. How can we achieve this kind of behavior using python smpplib.
Thanks

with regards
Sachin Maharjan

**Comment by eigenein** [comment, marked as answer]
Hi @lujamaharjan,

This is achieved via specifying the NPI and TON parameters:

```python
smpp_client.send_message(
    source_addr_ton=smpplib.consts.SMPP_TON_ALNUM,
    source_addr_npi=smpplib.consts.SMPP_NPI_UNK,
    source_addr="BANK_ALERT",
    …
)
```

For detailed explaination please refer to the SMPP specs: http://opensmpp.org/specs/smppv50.pdf (4.2.6.1.2, Alphanumeric Format).
